diskqueue.go

```diff
@@ -394,6 +432,17 @@ func (d *diskQueue) writeOne(data []byte) error {
 		return err
 	}
 
+	totalBytes := int64(4 + dataLen)
+
+	// check if we reached the file size limit with this message
+	if d.diskLimitFeatIsOn && d.writePos+totalBytes+8 >= d.maxBytesPerFile {
```

Comment:
looks like the checks (together with line 459 and 464) can be cleaned up a bit for better readability.

The block of `line 464 if fileSize >= d.maxBytesPerFile` is handling the case of finishing a file, so the number of messages can probably go into that block?

Also `d.maxBytesPerFile` is already reduced by 8 bytes in line 340, is it necessary to add another 8 bytes before checking?

Reply:
Nice catch with line 340! I will remove the addition of 8 bytes when doing the checks.

With regards to writing the number of messages in the block starting on line 464, I think it would be better to add it to the write buffer, which contains the write msg len and msg. I wanted to follow their principle of only writing to file once. That way if an error occurs when writing the number of messages to the buffer, nothing would have been written. This ensures that either the msg len, msg, and num msgs are all written or none are written.

Reply:
LGTM. in that case we could add a comment here so the reader understand better.

also we consider `totalBytes` as the number of bytes in this write, we could increment it here and eliminate the block of `line 461 fileSize += 8`

Reply:
Hello Leon, sorry for the confusion. I am revisiting the code, and line 340 decrements `d.maxBytesPerFileRead` which is used when getting the size of the file to read in `readOne()`. This is separate from `d.maxBytesPerFile`, thus the incrementation of the 8 bytes is necessary in the checks in `writeOne()` such as line 438. Incrementing `totalBytes` can cause an error, since totalBytes represents the total bytes of data excluding the number of messages data. Increasing totalBytes by 8 will increase writePos, which can increase `maxBytesPerFileRead` by 8. However, we do not want DiskQueue to read the number of messages data unless we are removing a file we did not finish reading.

Reply:
thanks for the explanation-) I was a big confused in the beginnning.

LGTM